SiameseDataset.py

```diff
@@ -0,0 +1,271 @@
+from __future__ import print_function
+import collections
+import h5py
+import numpy.random as random
+import numpy as np
+
+from CachedDataset2 import CachedDataset2
+from HDFDataset import FeatureSequenceStreamParser, SparseStreamParser, SegmentAlignmentStreamParser
+from Dataset import DatasetSeq
+
+
+
+class SiameseHDFDataset(CachedDataset2):
+  """
+  SiameseHDFDataset class allows to do sequence sampling for weakly-supervised training
+  It accepts data in the format of NextGenHDFDataset and performs sampling of sequence triplets before each epoch
+  Here we assume that each dataset from the input .hdf has a single label
+  Sampled triplets are splitted into different streams
+  In the config we can access streams by e.g. ["data:features_0", "data:features_1", "data:features_2"],
+  "data:features_0" stands for features of all sequences from the input data,
+  "data:features_1" is a stream of pairs, where for each sequence from "data:features_0" SiameseHDFDataset randomly samples a sequence with the same label
```

Comment:
I don't really understand what you mean by "stream of pairs". Can you specify that?
Esp, specify the dtype (I guess int32?), and the shape (is it like (num_sampled_sequences, 2)?).